docs/rules/prefer-user-event.md

```diff
@@ -0,0 +1,126 @@
+
+> [...] it is becoming apparent the need to express user actions on a web page
+> using a higher-level abstraction than `fireEvent`
+
+`userEvent` adds related event calls from browsers to make tests more realistic than its counterpart `fireEvent`, which is a low-level api.  
+See the appendix at the end to check how are the events from `fireEvent` mapped to `userEvent`
+
+## Rule Details
+
+This rules enforces the usage of [userEvent](https://github.com/testing-library/user-event) methods over `fireEvent`. By default, the methods from `userEvent` takes precedence, but you add exceptions by configuring the rule in `.eslintrc`
+
+See below for examples of valid usages of `fireEvent` methods with the configuration.
+
+Examples of **incorrect** code for this rule:
+
+```ts
+// a method in fireEvent that has a userEvent equivalent
+import { fireEvent } from '@testing-library/dom';
+fireEvent.click(node);
+
+// using fireEvent with an alias
+import { fireEvent as fireEventAliased } from '@testing-library/dom';
+fireEventAliased.click(node);
+
+// using fireEvent after importing the entire library
+import * as dom from '@testing-library/dom';
+dom.fireEvent.click(node);
+```
+
+Examples of **correct** code for this rule:
+
+```ts
+// any userEvent method
+userEvent.click();
+// fireEvent method that does not have an alternative in userEvent
+fireEvent.cut(node);
+import * as dom from '@testing-library/dom';
+dom.fireEvent.cut(node);
+
+// a function called fireEvent that's not imported from testing-library
+function fireEvent() {
+  // do stuff
+}
+fireEvent();
+```
+
+#### Options
+
+This rule allows to exclude specific functions with an equivalent in `userEvent` through configuration. This is useful if you need to allow an event from `fireEvent` to be used in the solution. For specific scenarios, you might want to consider disabling the rule inline.
+
+The configuration consists of an array of strings with the names of fireEvents methods to be excluded.  
+An example looks like this
+
+```json
+{
+  "rules": {
+    "prefer-user-event": [
+      "error",
+      {
+        "allowedMethods": ["click", "change"]
+      }
+    ]
+  }
+}
+```
+
+With this configuration example, the following use cases are considered valid
+
+```ts
+// using a named import
+import { fireEvent } from '@testing-library/dom';
+fireEvent.click(node);
+fireEvent.change(node, { target: { value: 'foo' } });
+
+// using fireEvent with an alias
+import { fireEvent as fireEventAliased } from '@testing-library/dom';
+fireEventAliased.click(node);
+fireEventAliased.change(node, { target: { value: 'foo' } });
+
+// using fireEvent after importing the entire library
+import * as dom from '@testing-library/dom';
+dom.fireEvent.click(node);
+dom.fireEvent.change(node, { target: { value: 'foo' } });
+```
+
+## When Not To Use It
+
+When you don't want to encourage the usage of `userEvent`
```

Comment:
I did not know what to put here 😅  🤷

Reply:
Maybe simplify this a little

```suggestion
When you don't want to use `userEvent`, such as if a legacy codebase is still using `fireEvent` or you need to have more low-level control over firing events (rather than the recommended approach of testing from a user's perspective)
```